apps/src/music/views/MusicView.jsx

```diff
@@ -359,11 +365,15 @@ class UnconnectedMusicView extends React.Component {
         levelToolboxDefinition
       );
     } else if (isEditingExemplar || isViewingExemplar) {
-      this.loadCode(this.getExemplarSources() || this.getStartSources());
-    } else if (this.getStartSources() || initialSources) {
-      const startSources = this.getStartSources();
+      this.loadCode(this.getExemplarSources() || startSources);
+    } else if (startSources || initialSources) {
       let codeToLoad = startSources;
-      if (initialSources?.source) {
+      if (
+        initialSources?.source &&
+        // Predict levels only use sources loaded from the server if the code is editable
+        // after submit, otherwise use the start sources.
+        (isPredictLevel ? codeEditableAfterSubmit : true)
```

Comment:
nit: may be easier to follow with something like
```
const isEditablePredictLevel = isPredictLevel && codeEditableAfterSubmit;

...

// Use initial sources if we have them, and we're not on a predict level, 
// or if we are, but the code is editable after submit. Otherwise, use start sources.
if (initialSources?.source && (!isPredictLevel || isEditablePredictLevel)) { ... }
```

Reply:
Yup, that looks good. I wrote this 5 different ways and couldn't quite come up with a way that I really liked.